backends/ilang/ilang_backend.cc

```diff
@@ -285,8 +285,19 @@ void ILANG_BACKEND::dump_module(std::ostream &f, std::string indent, RTLIL::Modu
 		if (!module->avail_parameters.empty()) {
 			if (only_selected)
 				f << stringf("\n");
-			for (auto &p : module->avail_parameters)
+			for (auto &p : module->avail_parameters) {
+
+				if (module->parameter_attributes.count(p)) {
+					f << stringf("\n");
+					for (auto& attr : module->parameter_attributes.at(p)) {
```

Comment:
I see you changed the ilang backend, but not the front end.  Did you test whether the frontend could accept this new output?

Reply:
I made corresponding changes to the frontend in `frontends/ilang/ilang_parser.y`. But I'll double check tomorrow if there are no issues with it.